crates/typst/src/foundations/int.rs

```diff
@@ -222,6 +263,23 @@ impl Repr for i64 {
     }
 }
 
+/// The endianness of a bytes to and from int conversion.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+enum Endianness {
+    Big,
+    Little,
+}
+
+cast! {
+    Endianness,
+    v: Str =>
+        match v.as_str() {
+            "big" => Ok(Self::Big),
+            "little" => Ok(Self::Little),
+            _ => Err("invalid endianness"),
+        }?,
```

Comment:
I think we can implement `FromStr` for `Endianness`, and call `Endianness::from_str` here. Using `FromStr` here might be more friendly to developers.

Reply:
We should rather use the `Cast` proc macro like for `enum Linebreaks`. Then, the docs will render a [nice table](https://typst.app/docs/reference/model/par/#parameters-linebreaks) completely auotmatically with the doc comments from the enum.

Reply:
Done in the latest commit. Didn't know that ty!